lib/spack/spack/detection/path.py

```diff
@@ -122,22 +129,11 @@ def _group_by_prefix(paths: Set[str]) -> Dict[str, Set[str]]:
     return groups
 
 
-class Finder:
+class Finder(spack.subprocess_context.TestState):
```

Comment:
@johnwparent Can we find a solution that doesn't require to modify an inner class just for testing? I'm happy to help with that in case.

Reply:
@scheibelp and I had discussed and landed on this approach (we do this in the install process as well) but I'd be happy to discuss alternatives.

AFAICT the only way to transparently propagate the monkeypatches to a subprocess only during test time would be either to monkeypatch the `Finder` class or modify the subprocess creation/pickling process. Am I missing something?

Reply:
Summarizing a discussion in DMs on Slack. We can try splitting tests on parent / child boundaries so to never have to spawn a process when we use monkey patching. For instance, the:
```
test_packages_yaml_format
```
test might just check that `spack.detection.update_configuration` works correctly for some well-defined input.